### DISCUSSION TITLE
Add the ability to configure KNX settings programmatically

### DISCUSSION THREAD

**Discussion by PaoloTK** [OpenKNX/OpenKNX#7] [General]
It would be beneficial for some use cases to configure a DIY KNX device programmatically rather than through ETS. For example, I would like to use the knx library on an ESP32 to integrate KNX in a third party software called WLED. It would make sense to be able to configure the device address and groups through WLED GUI, rather than having to make users use ETS. This was asked in the original repo but the maintainer said there's no plan to do so (https://github.com/thelsing/knx/issues/147).
I was wondering if the maintainers in this repo think it's something worth exploring?
I am interested in actively working on the feature, however the project is quite complex for me and having someone to follow along and ask questions to would be great.

**Comment by Ing-Dom** [comment]
OpenKNX aims to create devices that are configurable with ETS and as standard compatible as possible.
So, there is no sense in implementing such a way.
There are lightweight knx libs that do exactly that.

**Reply by PaoloTK** [reply]
First of all thank you for the quick reply.

> OpenKNX aims to create devices that are configurable with ETS and as standard compatible as possible. So, there is no sense in implementing such a way. There are lightweight knx libs that do exactly that.

Having the ability to offer both would be valuable in my opinion. After all it's not like there aren't KNX devices that allow you to change settings through means other than ETS (e.g. web server). 

However for my use case the license issue is a show stopper, and something I hadn't considered.

Thank you again.

**Comment by Ing-Dom** [comment]
you cant integrate this lib in WLED anyway, because WLED is MIT license and knx stack is GPL - you would have to change the WLED license to GPL then.
